src/corehost/cli/fxr/fx_ver.cpp

```diff
@@ -175,20 +299,35 @@ bool parse_internal(const pal::string_t& ver, fx_ver_t* fx_ver, bool parse_only_
     {
         return false;
     }
+    if (pat_sep - pat_start > 1 && ver[pat_start] == _X('0'))
+    {
+        return false;
+    }
 
     size_t pre_start = pat_sep;
-    size_t pre_sep = ver.find(_X('+'), pre_start);
-    if (pre_sep == pal::string_t::npos)
+    size_t pre_sep = ver.find(_X('+'), pat_sep);
+
+    pal::string_t pre = (pre_sep == pal::string_t::npos) ? ver.substr(pre_start) : ver.substr(pre_start, pre_sep - pre_start);
+
+    if (!validIdentifiers(pre))
```

Comment:
If I read this correctly it will parse for example `1.0.0-` as a valid version... which I think is not the case per the spec (although the spec is not very explicit about that).

Same goes for the build below.